File: HeYuanERP_Server/src/HeYuanERP.Domain/Entities/Account.cs
namespace HeYuanERP.Domain.Entities;

// 领域实体：客户/往来单位（Account）
// 对齐 OpenAPI：id, code, name, ownerId, taxNo, active, lastEventDate
// 说明：
// - 与联系人（Contact）一对多
// - 后续将通过配置类限制唯一索引与长度
public class Account
{
    // 主键
    public string Id { get; set; } = Guid.NewGuid().ToString("N");

    // 基本信息
    public string Code { get; set; } = string.Empty;   // 客户编码（唯一）
    public string Name { get; set; } = string.Empty;   // 客户名称
    public string? OwnerId { get; set; }               // 负责人（用户 Id）
    public string? TaxNo { get; set; }                 // 税号
    public bool Active { get; set; } = true;           // 是否启用
    public DateTime? LastEventDate { get; set; }       // 最近业务日期
    public string? Description { get; set; }           // 备注

    // 审计字段
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string? CreatedBy { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }

    // 导航：联系人
    public ICollection<Contact> Contacts { get; set; } = new List<Contact>();
}

File: heyuan-erp-web/src/types/account.ts
// 账户（客户）模块的前端类型定义
// 与后端 DTO 对齐，便于接口返回的类型提示与约束

// 通用分页结构（与服务端一致）
export interface Pagination<T> {
  items: T[];
  page: number;
  size: number;
  total: number;
}

// 查询参数
export interface AccountListQuery {
  page?: number;
  size?: number;
  keyword?: string;
  active?: boolean;
  ownerId?: string;
}

// 列表项
export interface AccountListItem {
  id: string;
  code: string;
  name: string;
  ownerId?: string;
  active: boolean;
  lastEventDate?: string; // ISO 字符串
  description?: string;
}

// 详情
export interface AccountDetail {
  id: string;
  code: string;
  name: string;
  ownerId?: string;
  taxNo?: string;
  active: boolean;
  lastEventDate?: string;
  description?: string;
  createdAt: string;
  createdBy?: string;
  updatedAt?: string;
  updatedBy?: string;
}

// 新建/编辑输入
export interface AccountCreateInput {
  code: string;
  name: string;
  ownerId?: string;
  taxNo?: string;
  active?: boolean;
  description?: string;
}

export interface AccountUpdateInput {
  code?: string;
  name: string;
  ownerId?: string;
  taxNo?: string;
  active?: boolean;
  description?: string;
}

// 共享/转移
export interface AccountShareRequest {
  targetUserId: string;
  permission: 'read' | 'edit';
  expireAt?: string; // ISO
}

export interface AccountTransferRequest {
  newOwnerId: string;
}

// 拜访
export interface AccountVisitCreate {
  visitDate?: string; // ISO
  contactId?: string;
  visitorId?: string;
  subject?: string;
  content?: string;
  location?: string;
  result?: string;
  nextActionAt?: string; // ISO
}

export interface AccountVisit {
  id: string;
  accountId: string;
  contactId?: string;
  visitorId?: string;
  visitDate: string;
  subject?: string;
  content?: string;
  location?: string;
  result?: string;
  nextActionAt?: string;
  createdAt: string;
  createdBy?: string;
  updatedAt?: string;
  updatedBy?: string;
}

// 附件（仅列表展示）
export interface AttachmentItem {
  id: string;
  fileName: string;
  contentType?: string;
  size: number;
  storageUri: string;
  uploadedAt: string;
  uploadedBy?: string;
}

